### ISSUE TITLE
[BUG] Scorer freezes

### ISSUE THREAD

**Issue by N0Dr4m4L4m4** [takuseno/d3rlpy#260]
**Describe the bug**
I am trying to run the d3rl lib with online learning, however while the evaluation the loop seems to freeze,because the prediction is not quiet well. Means following:
```
def scorer(algo: AlgoProtocol, *args: Any) -> float:
        if is_image:
            stacked_observation = StackedObservation(
                observation_shape, algo.n_frames
            )
        episode_rewards = []
        for _ in range(n_trials):
            observation = env.reset()
            episode_reward = 0.0

            # frame stacking
            if is_image:
                stacked_observation.clear()
                stacked_observation.append(observation)

            while True:
                # take action
                if np.random.random() < epsilon:
                    action = env.action_space.sample()
                else:
                    if is_image:
                        action = algo.predict([stacked_observation.eval()])[0]
                    else:
                        action = algo.predict([observation])[0]
                        print(action)

                observation, reward, done, info = env.step(action)
                episode_reward += reward

                if is_image:
                    stacked_observation.append(observation)

                if render:
                    env.render()
                
                if done:
                    break
                
            episode_rewards.append(episode_reward)
        return float(np.mean(episode_rewards))

    return scorer
``` 
Problem here, there is a while loop with true and it just terminates if, the algo makes dicisions. If not it will stay forever in this loops. 

**To Reproduce**
Use an custom env for gym 

**Expected behavior**
After some time it should be exit and not stay forever in while true

**Additional context**
**if done** never reachs true in my case, is that the indendet behavior ?

**Comment by takuseno** [comment]
Hi @N0Dr4m4L4m4 . This is not a bug of d3rlpy. In your custom env, you need to implement timeout to exit.

**Comment by N0Dr4m4L4m4** [comment]
Hey there,


Is there any advice or description for that?

**Comment by takuseno** [comment]
It's difificult to say because completely depending on environments. But the time until timeout should be long enough for agents to complete tasks.

**Comment by takuseno** [comment]
Feel free to reopen this if you have any further discussion. Please note that the nightly branch has been merged and the latest update `v2.0.2` has been released.
